chef-config/lib/chef-config/windows.rb

```diff
@@ -16,13 +16,18 @@
 #
 
 module ChefConfig
-
   def self.windows?
-    if RUBY_PLATFORM =~ /mswin|mingw|windows/
+    @node ||= nil
+    if @node && @node[:os]
```

Comment:
Wouldn't this always result in returning true if @node is present and has an :os key, even on non-windows systems?